mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -3861,6 +3876,28 @@ public DocumentationMember (XmlNode node)
 			for (int i = 0; i < p.Count; ++i)
 				Parameters.Add (p [i].Attributes ["Type"].Value);
 		}
+		XmlNodeList tp = node.SelectNodes ("TypeParameters/TypeParameter[not(@apistyle) or @apistyle='classic']");
+		if (tp.Count > 0) {
+			TypeParameters = new StringList (tp.Count);
+			for (int i = 0; i < tp.Count; ++i)
+				TypeParameters.Add (tp [i].Attributes ["Name"].Value);
+		}
+		else {
+			DiscernTypeParameters ();
```

Comment:
It would probably be a good "sanity check" to _always_ call `DiscernTypeParameters()` (-ish...), and if `TypeParameters` already has elements, ensure that the "discerned" count matches the "real" count, because if they don't match, something has gone horribly, horribly, wrong.